internal/commands/config/config.go

```diff
@@ -0,0 +1,56 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/cirruslabs/gitlab-tart-executor/internal/tart"
+	"github.com/spf13/cobra"
+	"os"
+)
+
+func NewCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:   "config",
+		Short: "Configure GitLab Runner",
+		RunE:  runConfig,
+	}
+
+	return command
+}
+
+func runConfig(cmd *cobra.Command, args []string) error {
+	gitlabRunnerConfig := struct {
+		BuildsDir string            `json:"builds_dir"`
+		CacheDir  string            `json:"cache_dir"`
+		JobEnv    map[string]string `json:"job_env,omitempty"`
+	}{
+		BuildsDir: "builds",
+		CacheDir:  "cache",
+		JobEnv:    map[string]string{},
+	}
+
+	tartConfig, err := tart.NewConfigFromEnvironment()
+	if err != nil {
+		return err
+	}
+
+	if tartConfig.HostDir {
+		gitlabRunnerConfig.BuildsDir = "/Volumes/My Shared Files/hostdir"
+
+		tmpDir, err := os.MkdirTemp("", "")
```

Comment:
What do you think of just deterministically getting it via job id instead of playing smart and passing it around via a new environment variable?

Reply:
I'm not a fan of this approach.

If it's possible to use a standard library function that (1) figures out the base temporary directory for you (2) guarantees no collisions and then pass the resulting folder using a standard mechanism, which [GitLab advertises it as such](https://docs.gitlab.com/runner/executors/custom.html#config), why not go for it?

Reply:
I don't see where GitLab advertises it. Following the link I see build directory being pretty deterministic e.g. `"/builds/${CUSTOM_ENV_CI_CONCURRENT_PROJECT_ID}/${CUSTOM_ENV_CI_PROJECT_PATH_SLUG}"`

Reply:
> I don't see where GitLab advertises it. Following the link I see build directory being pretty deterministic e.g. `"/builds/${CUSTOM_ENV_CI_CONCURRENT_PROJECT_ID}/${CUSTOM_ENV_CI_PROJECT_PATH_SLUG}"`

My bad, I've got the link wrong, [here it is](https://docs.gitlab.com/runner/executors/custom.html#job_env-usage):

>The main purpose of `job_env` configuration is to pass variables **to the context of custom executor driver calls** for subsequent stages of the job execution.

The example with `"/builds/${CUSTOM_ENV_CI_CONCURRENT_PROJECT_ID}/${CUSTOM_ENV_CI_PROJECT_PATH_SLUG}"` is OK for cases when you can't use anything other than pre-defined environment variables, but we can, so why not simply create a temporary directory instead and pass it to the other stages?

Reply:
I honestly do not see any cons of using this approach. I don't see how there can be a collision in real life usage. IMO it brings unnecessary complexity by introducing a state and putting some large folders randomly on disk that potentially can not being deleted properly.

Having a deterministic path like `/tmp/tart-executor-build-dir-$JOB_ID` just makes things easier: no need to pass a state, you can debug each state from an IDE and don't worry about cleaning up some random folders, in case folders won't be cleaned up because of a potential bug in the executor itself, it will be easy to determine when it happened and get logs.

Reply:
See https://github.com/cirruslabs/gitlab-tart-executor/pull/11/commits/dc2603d23c1e4aeffb21c033cb1099d3d1ee694d.